src/Models/Booking.php

```diff
@@ -5,29 +5,39 @@
 namespace Tipoff\Bookings\Models;
 
 use Carbon\Carbon;
+use Tipoff\Support\Contracts\Checkout\CartInterface;
 use Tipoff\Support\Models\BaseModel;
 use Tipoff\Support\Traits\HasCreator;
 use Tipoff\Support\Traits\HasPackageFactory;
 use Tipoff\Support\Traits\HasUpdater;
 
-class Booking extends BaseModel
+class Booking extends BaseModel implements CartInterface
 {
     use HasPackageFactory;
     use HasCreator;
     use HasUpdater;
 
     protected $with = [
-        'variation',
+        'rate',
         'experience',
         'order',
         'agent',
         'user',
         'subject',
     ];
 
-    public function computeTaxes()
+    /**
+     * Generate amount, total_taxes and total_fees.
+     */
+    public function generatePricing()
     {
-        return $this->amount * ($this->tax->percent / 100);
+        if (empty($booking->slot_number)) {
+            throw new \Exception('A booking must be for an availability slot.');
+        }
+
+        $this->total_taxes = $this->getTax();
+        $this->total_amount = $this->getItemAmountTotal();
+        // $this->fees = '';
```

Comment:
Fees were empty in the original method. What should be used to determine this? What separates fee from tax?

Reply:
same like in case of tax